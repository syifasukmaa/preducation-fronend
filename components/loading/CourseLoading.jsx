import React from 'react';

export default function CourseLoading() {
  return (
    <tr className="my-10">
      <td className="w-12 px-4">
        <p className="py-1 my-4 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="w-24 px-4">
        <div className="flex flex-col justify-between h-full">
          <p className="w-1/2 py-1 rounded-sm animate-pulse bg-neutral-03"></p>
          <p className="py-1 mt-[3px] w-3/4 animate-pulse bg-neutral-03 rounded-sm"></p>
        </div>
      </td>
      <td className="px-4 w-72">
        <p className="py-1 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="w-12 px-4">
        <p className="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="w-16 px-4">
        <p className="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="w-16 px-4">
        <p className="py-1 rounded-sm animate-pulse bg-neutral-03"></p>
      </td>
      <td className="w-48 px-4">
        <div className="flex flex-col h-full lg:flex-row">
          <p className="w-16 py-2 mt-2 rounded-md xl:mt-0 animate-pulse bg-neutral-03"></p>
          <p className="w-16 py-2 mt-2 ml-2 rounded-md xl:mt-0 animate-pulse bg-neutral-03"></p>
        </div>
      </td>
    </tr>
  );
}
