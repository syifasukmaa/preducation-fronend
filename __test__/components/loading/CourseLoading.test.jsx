import CourseLoading from '@/components/loading/CourseLoading'
import { render } from '@testing-library/react'

describe('Course Loading', () => {
  it('renders loading skeleton', () => {
    const container = render(<CourseLoading />)
    expect(container).toMatchSnapshot()
  })
})
