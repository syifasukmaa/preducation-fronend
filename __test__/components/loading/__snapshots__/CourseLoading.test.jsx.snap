// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Loading renders loading skeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="my-10"
      >
        <td
          class="w-12 px-4"
        >
          <p
            class="py-1 my-4 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="w-24 px-4"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="w-1/2 py-1 rounded-sm animate-pulse bg-neutral-03"
            />
            <p
              class="py-1 mt-[3px] w-3/4 animate-pulse bg-neutral-03 rounded-sm"
            />
          </div>
        </td>
        <td
          class="px-4 w-72"
        >
          <p
            class="py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="w-12 px-4"
        >
          <p
            class="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="w-16 px-4"
        >
          <p
            class="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="w-16 px-4"
        >
          <p
            class="py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="w-48 px-4"
        >
          <div
            class="flex flex-col h-full lg:flex-row"
          >
            <p
              class="w-20 py-4 mt-2 rounded-lg xl:mt-0 animate-pulse bg-neutral-03"
            />
            <p
              class="w-20 py-4 mt-2 ml-2 rounded-lg xl:mt-0 animate-pulse bg-neutral-03"
            />
          </div>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="my-10"
    >
      <td
        class="w-12 px-4"
      >
        <p
          class="py-1 my-4 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="w-24 px-4"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="w-1/2 py-1 rounded-sm animate-pulse bg-neutral-03"
          />
          <p
            class="py-1 mt-[3px] w-3/4 animate-pulse bg-neutral-03 rounded-sm"
          />
        </div>
      </td>
      <td
        class="px-4 w-72"
      >
        <p
          class="py-1 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="w-12 px-4"
      >
        <p
          class="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="w-16 px-4"
      >
        <p
          class="px-4 py-1 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="w-16 px-4"
      >
        <p
          class="py-1 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="w-48 px-4"
      >
        <div
          class="flex flex-col h-full lg:flex-row"
        >
          <p
            class="w-20 py-4 mt-2 rounded-lg xl:mt-0 animate-pulse bg-neutral-03"
          />
          <p
            class="w-20 py-4 mt-2 ml-2 rounded-lg xl:mt-0 animate-pulse bg-neutral-03"
          />
        </div>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
